### PR TITLE
Feature/tao 2914 metadata export config

### PR DESCRIPTION
**Pull request by siwane**
To test it use client extension with metadata injector/extractor/classlookup and guardians
Try to import an item package and check metadata import
Don't forget to check the upgrade of taoQtiItem to check retrocompatibility

### CODE REVIEW

File: model/qti/metadata/AbstractMetadataService.php [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+        switch ($key) {
+            case self::EXTRACTOR_KEY:
+                $this->registerInstance(self::EXTRACTOR_KEY, $name, MetadataExtractor::class);
+                break;
+            case self::INJECTOR_KEY:
+                $this->registerInstance(self::INJECTOR_KEY, $name, MetadataInjector::class);
+                break;
+            default:
+                throw new \common_Exception(__('Unknow $key to register MetadataService instance'));
```
Comment (bugalot) [inline model/qti/metadata/AbstractMetadataService.php]: UNKNOWN :) !

Comment (siwane) [inline model/qti/metadata/AbstractMetadataService.php]: done

File: model/qti/metadata/MetadataRegistry.php [hunk 2]
```diff
@@ -80,43 +84,49 @@ protected function getExtension()
...
-            
-            return array('injectors' => array(), 'extractors' => array(), 'guardians' => array(), 'classLookups' => array());
+        $instances = $this->getServiceManager()
+            ->get(MetadataService::SERVICE_ID)
+            ->getImporter()
+            ->getOptions();
+
+        $mapping = [];
+        foreach ($instances as $key => $instance) {
+            $mapping[$key][] = get_class($instances);
```
Comment (bugalot) [inline model/qti/metadata/MetadataRegistry.php]: get_class($instances) or get_class($instance) ?

**Comment by siwane** [comment]
@bugalot PR updated

**Pull request merged**